presto-common/src/main/java/com/facebook/presto/common/zorder/ZOrderIntegers.java

```diff
@@ -0,0 +1,169 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.common.zorder;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static java.lang.String.format;
+
+public class ZOrderIntegers
```

Comment:
Do we expect other types than integers? If not, we don't have to create `ZOrder` interface. We can just call this class `ZOrder`

Reply:
If I have time, I'll possibly be implementing ZOrderFloats and/or ZOrderStrings. 

> always make `List` with specific types `List<Integer>`

In that case, the type of List being returned by classes that implement ZOrder will be different.

For now, though, I can remove the ZOrder interface and rename ZOrderIntegers to ZOrder for this diff. Would that work?

Reply:
Yeah, for now let us remove the ZOrder class. Thanks.